src/main/webapp/app/exercises/programming/assess/code-editor-tutor-assessment-container.component.ts

```diff
@@ -386,19 +380,14 @@ export class CodeEditorTutorAssessmentContainerComponent implements OnInit, OnDe
     }
 
     private handleFeedback(): void {
-        // Setup automatic feedback
-        this.automaticFeedback = this.automaticResult?.feedbacks ?? [];
-        this.automaticFeedback.forEach((feedback) => {
-            feedback.id = undefined;
-            if (!feedback.credits) {
-                feedback.credits = 0;
-            }
-        });
-
-        // Setup not automatically generated feedbacks
         const feedbacks = this.manualResult?.feedbacks || [];
-        this.unreferencedFeedback = feedbacks.filter((feedbackElement) => feedbackElement.reference == undefined && feedbackElement.type === FeedbackType.MANUAL_UNREFERENCED);
+        this.automaticFeedback = feedbacks.filter((feedback) => feedback.type === FeedbackType.AUTOMATIC);
+        // When manual result only contains automatic feedback elements (when assessing for the first time), no manual assessment was yet saved or submitted.
+        if (this.manualResult?.feedbacks && this.manualResult.feedbacks.length === this.automaticFeedback.length) {
```

Comment:
I would rather check here, that no manual feedback exist.

Reply:
Discussed and decided to keep it like it is 👍